openstack/containerinfra/v1/clustertemplates/requests.go

```diff
@@ -123,3 +124,51 @@ func Get(client *gophercloud.ServiceClient, id string) (r GetResult) {
 	}
 	return
 }
+
+type UpdateOpts struct {
+	Path  string `json:"path" required:"true"`
+	Value string `json:"value,omitempty"`
+	Op    string `json:"op" required:"true"`
+}
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToClusterTemplateUpdateMap() (map[string]interface{}, error)
+}
+
+// ToClusterUpdateMap assembles a request body based on the contents of
+// UpdateOpts.
+func (opts UpdateOpts) ToClusterTemplateUpdateMap() (map[string]interface{}, error) {
+	if opts.Op != "remove" && opts.Value == "" {
+		return nil, fmt.Errorf("Value field must be provied for Op=[%s]. Only when Op=remove can have empty Value field", opts.Op)
+	}
```

Comment:
Will the API return an error if an illegal combination is given? If so, let's just let the API error be returned to the user rather than masking it with our own error.

Reply:
Will fix and let back-end handle validation rather than our own masking.